test/integration/operator_policy_error_test.go

```diff
@@ -0,0 +1,193 @@
+			)
+			Expect(managedPolicy).NotTo(BeNil())
+		})
+
+		It(operatorPolicyName+" should be NonCompliant due to invalid namespace", func() {
+			By("Checking if the status of the operator policy is NonCompliant")
+			Eventually(
+				common.GetComplianceState(policyName),
+				defaultTimeoutSeconds*2,
+				1,
+			).Should(Equal(policiesv1.NonCompliant))
+			Eventually(
+				common.GetOpPolicyCompMsg(operatorPolicyName),
+				defaultTimeoutSeconds*2,
+				1,
+			).Should(ContainSubstring("the operator namespace ('grcqeoptest-notcreated') does not exist"))
+		})
+
+		It("Namespace "+subscriptionNamespace+" should be created", func() {
+			_, err := common.OcManaged(
+				"create", "namespace", subscriptionNamespace,
+			)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It(operatorPolicyName+" should be patched with correct namespace", func() {
+			_, err := common.OcHub(
+				"patch", "policies.policy.open-cluster-management.io", policyName,
+				"-n", userNamespace, "--type=json", "-p", `[{
+				"op":"replace", 
+				"path":"/spec/policy-templates/0/objectDefinition/spec/subscription/namespace",
+				"value":"grcqeoptest-ns-43568"
+			}]`)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It(operatorPolicyName+" should be patched with incorrect source", func() {
+			_, err := common.OcHub(
+				"patch", "policies.policy.open-cluster-management.io", policyName,
+				"-n", userNamespace, "--type=json", "-p", `[{
+				"op":"replace", 
+				"path":"/spec/policy-templates/0/objectDefinition/spec/subscription/source",
+				"value":"redhat-operators-invalid"
+			}]`)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It(operatorPolicyName+" should be NonCompliant due to incorrect source", func() {
+			By("Checking if the status of the operator policy is NonCompliant")
+			Eventually(
+				common.GetComplianceState(policyName),
+				defaultTimeoutSeconds*2,
+				1,
+			).Should(Equal(policiesv1.NonCompliant))
+			Eventually(
+				common.GetOpPolicyCompMsg(operatorPolicyName),
+				defaultTimeoutSeconds*2,
+				1,
+			).Should(ContainSubstring("CatalogSource 'redhat-operators-invalid' was not found"))
+		})
+
+		It(operatorPolicyName+" should be patched with correct source", func() {
+			_, err := common.OcHub(
+				"patch", "policies.policy.open-cluster-management.io", policyName,
+				"-n", userNamespace, "--type=json", "-p", `[{
+				"op":"replace", 
+				"path":"/spec/policy-templates/0/objectDefinition/spec/subscription/source",
+				"value":"redhat-operators"
+			}]`)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It(operatorPolicyName+" should be patched with incorrect installPlanApproval", func() {
+			_, err := common.OcHub(
+				"patch", "policies.policy.open-cluster-management.io", policyName,
+				"-n", userNamespace, "--type=json", "-p", `[{
+				"op":"replace", 
+				"path":"/spec/policy-templates/0/objectDefinition/spec/subscription/installPlanApproval",
+				"value":"Invalid"
+			}]`)
+			Expect(err).ToNot(HaveOccurred())
+		})
```

Comment:
Between checking the status for something specific, it looks like this suite patches the policy twice: once to undo a previous change, then it applies a new detail to check. It seems like it could be slightly nicer to replace that first patch with an "apply" back to the known state. That way, if one of the patch commands is adjusted in the future, it's corresponding "undo" doesn't need to be updated - the "apply" will always work and set the policy to something known.